pkg/operator/deploy/deploy.go

```diff
@@ -0,0 +1,173 @@
+)
+
+var (
+	manifestLocation       = homedir.HomeDir() + "/go/src/github.com/operator-framework/operator-metering/manifests/deploy/"
+	upstreamManifestPath   = manifestLocation + "upstream/" + manifestAnsibleOperator
+	openshiftManifestPath  = manifestLocation + "openshift/" + manifestAnsibleOperator
+	ocpTestingManifestPath = manifestLocation + "ocp-testing/" + manifestAnsibleOperator
+)
+
+const (
+	defaultMeteringConfig   = "meteringconfig.yaml"
+	manifestAnsibleOperator = "metering-ansible-operator/"
+
+	meteringDeploymentFile         = "metering-operator-deployment.yaml"
+	meteringServiceAccountFile     = "metering-operator-service-account.yaml"
+	meteringRoleBindingFile        = "metering-operator-rolebinding.yaml"
+	meteringRoleFile               = "metering-operator-role.yaml"
+	meteringClusterRoleBindingFile = "metering-operator-clusterrolebinding.yaml"
+	meteringClusterRoleFile        = "metering-operator-clusterrole.yaml"
+
+	hivetableFile        = "hive.crd.yaml"
+	prestotableFile      = "prestotable.crd.yaml"
+	meteringconfigFile   = "meteringconfig.crd.yaml"
+	reportFile           = "report.crd.yaml"
+	reportdatasourceFile = "reportdatasource.crd.yaml"
+	reportqueryFile      = "reportquery.crd.yaml"
+	storagelocationFile  = "storagelocation.crd.yaml"
+)
+
+// CRD is a structure that holds the information needed to install
+// these resources, like the name of the CRD, the path to the CRD
+// in the manifests directory, and a pointer to the apiextensions CRD type.
+type CRD struct {
+	Name string
+	Path string
+	CRD  *apiextv1beta1.CustomResourceDefinition
+}
+
+// Deploy is a structure that holds the information needed to handle
+// different metering deploy configurations, like the type of deploy
+// (e.g. install/uninstall), the deploy platform, deleting metering CRDs
+// or the namespace during an install, etc. This structure also contains fields
+// like the various clientsets used to manage metering resources in the deploy methods,
+// and a reference to an initialized logrus field logger.
+type Deploy struct {
+	Type                   string
+	Namespace              string
+	Platform               string
+	ManifestLocation       string
+	MeteringCR             string
+	SkipMeteringDeployment bool
+	DeleteCRDs             bool
+	DeleteCRB              bool
+	DeleteNamespace        bool
+	DeletePVCs             bool
+	Repo                   string
+	Tag                    string
+	TargetNamespaces       string
+	CRDs                   []CRD
+	Logger                 log.FieldLogger
+	Client                 *kubernetes.Clientset
+	APIExtClient           apiextclientv1beta1.CustomResourceDefinitionsGetter
+	MeteringClient         *metering.MeteringV1Client
+}
+
+// Driver handles the overarching workflow of deploying metering by calling on deploy helper methods/functions
+// that instantiate a Deploy structure and use the various clients to handle the managing the metering resources
+func (deploy *Deploy) Driver() error {
```

Comment:
I'm wondering if we need this or if we could just have install/uninstall be called instead, and make setup something they each call, or something that happens when you create the `Deploy` struct, perhaps in a constructor function.